### PR TITLE
Refactor/CP-1932 Topic View not openable issue 

### PR DESCRIPTION
**Pull request by HardikLakum**
I have solved the issue with the topic view not being openable after closing from the background

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 1]
```diff
@@ -2157,26 +2157,41 @@ public void showTopicsDialog(Context dialogActivity, TopicsDialogListener topics
...
                     }
 
+                    Set<String> selectedTopics = instance.getSubscriptionTopics();
+
                     if (hasDeSelectAll()) {
-                        setCheckboxList(parentLayout, checkboxDeSelectAll, channelTopics, checkedTopics, topicIds, true, nightModeFlags);
+                        setCheckboxList(parentLayout, checkboxDeSelectAll, channelTopics, checkedTopics, topicIds, true, nightModeFlags, selectedTopics);
                     } else {
-                        setCheckboxList(parentLayout, checkboxDeSelectAll, channelTopics, checkedTopics, topicIds, false, nightModeFlags);
+                        setCheckboxList(parentLayout, checkboxDeSelectAll, channelTopics, checkedTopics, topicIds, false, nightModeFlags, selectedTopics);
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2165]: If else can be simplified in one method call.

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 2]
```diff
@@ -2157,26 +2157,41 @@ public void showTopicsDialog(Context dialogActivity, TopicsDialogListener topics
...
                             unsubscribe();
                             setDeSelectAll(true);
+                            for (int j = 0; j < channelTopics.length(); j++) {
+                                try {
+                                    JSONObject channelTopic = (JSONObject) channelTopics.get(i);
+                                    channelTopic.put("defaultUnchecked", true);
+                                    selectedTopics.clear();
+                                    checkedTopics[i] = false;
+                                } catch (JSONException e) {
+                                    e.printStackTrace();
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2192]: Use Log. For logging

**Review by gbhrdt — CHANGES_REQUESTED** [review]
.

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 3]
```diff
@@ -2268,7 +2272,7 @@ private void setCheckboxList(LinearLayout parentLayout, CheckBox checkboxDeSelec
                         }
                     }
 
-                    if (!hasDeSelectAll()) {
+                    if (!hasDeSelectAll() && !isDeselectAll) {
```
Comment (YasinY) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2275]: do not write is or has in variables in java...

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 4]
```diff
@@ -2157,26 +2157,41 @@ public void showTopicsDialog(Context dialogActivity, TopicsDialogListener topics
...
                         showingTopicsDialog = false;
                     });
+
                     alertBuilder.setPositiveButton(CleverPush.context.getResources().getString(R.string.save), (dialogInterface, i) -> {
                         if (checkboxDeSelectAll.isChecked()) {
                             unsubscribe();
                             setDeSelectAll(true);
+                            for (int j = 0; j < channelTopics.length(); j++) {
+                                try {
+                                    JSONObject channelTopic = (JSONObject) channelTopics.get(i);
```
Comment (YasinY) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:2187]: looks like you could create a mapper for this.

**Review by YasinY — CHANGES_REQUESTED** [review]

**Review by YasinY — APPROVED** [review]

**Pull request merged**